hyperspy/drawing/spectrum.py

```diff
@@ -332,19 +332,18 @@ def update(self, force_replot=False):
         else:
             ydata = self.data_function(axes_manager=self.axes_manager).imag
 
-        new_axis = self.axes_manager.signal_axes[0].axis
-        if len(new_axis) != len(self.axis) or \
-           (len(new_axis) == len(self.axis) and
-                np.any(np.not_equal(new_axis, self.axis))):
-            self.axis = new_axis
-            self.ax.set_xlim(self.axis[0], self.axis[-1])
-            self.line.set_data(self.axis, ydata)
+        old_xaxis = self.line.get_xdata()
+        if len(old_xaxis) != self.axis.size or \
+           (len(old_xaxis) == self.axis.size and
```

Comment:
`len(old_xaxis) != self.axis.size or (len(old_xaxis) == self.axis.size and <something>`
Second size comparison (`==`) is redundant, right? Or am I missing something in the logic?